polygon/heimdall/span_block_producers_tracker.go

```diff
@@ -23,31 +23,32 @@ import (
 	"sync/atomic"
 
 	"github.com/erigontech/erigon-lib/log/v3"
-	"github.com/erigontech/erigon/polygon/bor/borcfg"
 	"github.com/erigontech/erigon/polygon/bor/valset"
 )
 
+type CalculateSprintNumberType func(uint64) uint64
```

Comment:
When initialising for standalone `rpcdaemon` we only have access to the `Bor` interface. Since we are using just one function from the config, I feel it is cleaner to extend the interface to return the function we need and simplify the constructor rather than typecasting to `*borcfg.BorConfig`.

In `bridge.Reader` above, we do a similar thing for `stateReceiverContractAddress`.

Reply:
ok, it seems like we are heading in a direction where at some point in the future we may end up having a `BridgeConfig` and `HeimdallConfig` which are an analogy to `BorConfig` but more compact and more relevant to the needs of each of the 2

Reply:
Yes, if we end up needing multiple pieces of data from borConfig, it would be ideal to have those.